apis/v1alpha1/gatewayclass_types.go

```diff
@@ -63,23 +63,25 @@ type GatewayClassSpec struct {
 	Controller string `json:"controller"`
 
 	// AllowedGatewayNamespaceSelector is a selector of namespaces that Gateways
-	// can use this GatewayClass from. This is a standard Kubernetes
-	// LabelSelector. Controllers must not support Gateways in namespaces
-	// outside this selector.
+	// of this class can be created in. Implementations must not support
+	// Gateways when they are created in namespaces prohibitied by this field.
 	//
-	// An empty selector (default) indicates that Gateways can use this
-	// GatewayClass from any namespace.
+	// This selector may be changed in a way that invalidates existing Gateways.
+	// Implementations may choose to immediately terminate related
+	// infrastructure. If implementations can differentiate between provisioned
+	// and pending Gateways, they may choose to continue to support any Gateways
+	// that were provisioned before this field changed. This differentiation can
+	// not be based on the Gateway status or other fields that users can
```

Comment:
@robscott why can't the differentiation between provisioned and pending be based on gateway status?

Reply:
It's a security issue. This field has no value if you can always bypass it by manually modifying the Gateway status.

Reply:
Added some clarification here, thanks!